File: frontend/types/quiz.ts
export interface QuizOption {
  id: string;
  label: string;
}

export interface QuizItem {
  id: string;
  stem: string;
  options: QuizOption[];
  multi: boolean;
  answer: string[];
  explanation?: string;
  type?: string;
  difficulty?: string;
  codeSnippet?: string | null;
}

export interface QuizSubmitAnswer {
  id: string;
  choices: string[];
}

export interface QuizSubmitRequest {
  topic: string;
  chapter?: string;
  answers: QuizSubmitAnswer[];
}

export interface QuizResult {
  score: number;
  total: number;
  correctIds: string[];
  wrongIds: string[];
  submittedAt: string;
  durationMs: number;
}

export interface QuizAnswerDetail {
  question_id: string; // 改为string以避免大整数精度问题
  is_correct: boolean;
  correct_answers: string[];
  explanation: string;
  type: string;       // 题型：single/multiple
  difficulty: string; // 难度：easy/medium/hard
}

export interface QuizSubmitResult {
  score: number;
  total_questions: number;
  correct_answers: number;
  passed: boolean;
  details: QuizAnswerDetail[];
}

export interface QuizQuestion {
  id: string; // 改为string以避免大整数精度问题
  type: string;
  difficulty: string;
  question: string;
  options: QuizOption[];
  codeSnippet?: string | null;
}

export interface QuizSessionPayload {
  sessionId: string;
  topic: string;
  chapter: string;
  questions: QuizQuestion[];
}

export interface QuizHistoryItem {
  id: number;
  sessionId?: string;
  topic: string;
  chapter?: string | null;
  score: number;
  totalQuestions?: number;
  correctAnswers?: number;
  passed?: boolean;
  completedAt?: string | null;
}

export interface QuizReviewMeta {
  sessionId: string;
  topic: string;
  chapter: string;
  score: number;
  passed: boolean;
  completedAt?: string | null;
}

export interface QuizReviewItem {
  questionId: number;
  stem: string;
  options: string[];
  userChoice: string;
  correctChoice: string;
  isCorrect: boolean;
  explanation: string;
  type: string;       // 题型：single/multiple
  difficulty: string; // 难度：easy/medium/hard
}

export interface QuizReviewDetail {
  meta: QuizReviewMeta;
  items: QuizReviewItem[];
}

export interface QuizStats {
  total: number;
  byType: Record<string, number>;
  byDifficulty: Record<string, number>;
}

File: backend/internal/domain/quiz/quiz.go
package quiz

import "time"

// QuizQuestion 描述题库中的单个题目。
type QuizQuestion struct {
	ID             int64     `json:"id" orm:"id"`
	Topic          string    `json:"topic" orm:"topic"`
	Chapter        string    `json:"chapter" orm:"chapter"`
	Type           string    `json:"type" orm:"type"`
	Difficulty     string    `json:"difficulty" orm:"difficulty"`
	Question       string    `json:"question" orm:"question"`
	Options        string    `json:"options" orm:"options"`
	CorrectAnswers string    `json:"correctAnswers" orm:"correct_answers"`
	Explanation    string    `json:"explanation" orm:"explanation"`
	CodeSnippet    *string   `json:"codeSnippet" orm:"code_snippet"`
	CreatedAt      time.Time `json:"createdAt" orm:"created_at"`
	UpdatedAt      time.Time `json:"updatedAt" orm:"updated_at"`
}

// QuizSession 记录一次测验的摘要信息。
type QuizSession struct {
	ID             int64      `json:"id" orm:"id"`
	SessionID      string     `json:"sessionId" orm:"session_id"`
	UserID         int64      `json:"userId" orm:"user_id"`
	Topic          string     `json:"topic" orm:"topic"`
	Chapter        string     `json:"chapter" orm:"chapter"`
	TotalQuestions int        `json:"totalQuestions" orm:"total_questions"`
	CorrectAnswers int        `json:"correctAnswers" orm:"correct_answers"`
	Score          int        `json:"score" orm:"score"`
	Passed         bool       `json:"passed" orm:"passed"`
	StartedAt      time.Time  `json:"startedAt" orm:"started_at"`
	CompletedAt    *time.Time `json:"completedAt" orm:"completed_at"`
	CreatedAt      time.Time  `json:"createdAt" orm:"created_at"`
}

// QuizAttempt 记录用户对单题的作答。
type QuizAttempt struct {
	ID          int64     `json:"id" orm:"id"`
	SessionID   string    `json:"sessionId" orm:"session_id"`
	UserID      int64     `json:"userId" orm:"user_id"`
	Topic       string    `json:"topic" orm:"topic"`
	Chapter     string    `json:"chapter" orm:"chapter"`
	QuestionID  int64     `json:"questionId" orm:"question_id"`
	UserAnswers string    `json:"userAnswers" orm:"user_answers"`
	IsCorrect   bool      `json:"isCorrect" orm:"is_correct"`
	AttemptedAt time.Time `json:"attemptedAt" orm:"attempted_at"`
}

// 题型与难度枚举。
const (
	QuestionTypeSingle         = "single"
	QuestionTypeMultiple       = "multiple"
	QuestionTypeTrueFalse      = "truefalse"
	QuestionTypeCodeOutput     = "code_output"
	QuestionTypeCodeCorrection = "code_correction"

	DifficultyEasy   = "easy"
	DifficultyMedium = "medium"
	DifficultyHard   = "hard"
)
